integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/GreenMailUtil.java

```diff
@@ -0,0 +1,308 @@
+            throws MessagingException, UserException, InterruptedException {
+        MimeMessage message = new MimeMessage((Session) null);
+        message.setFrom(new InternetAddress("some@mail.com"));
+        message.addRecipient(Message.RecipientType.TO, new InternetAddress("wso2@localhost"));
+        message.setSubject(subject);
+        message.setText("Content :" + subject);
+        message.setSentDate(new Date());
+        if (!headers.isEmpty()) {
+            for (Map.Entry<String, String> entry : headers.entrySet()) {
+                message.addHeader(entry.getKey(), entry.getValue());
+            }
+        }
+
+        greenMailUser.deliver(message);
+        Thread.sleep(10000);
+    }
+
+    /**
+     * Connecting to the pop3 mail store
+     * @return pop3 store connection
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    private static Store getPopConnection() throws ESBMailTransportIntegrationTestException {
+        Properties props = new Properties();
+        Session session = Session.getInstance(props);
+        URLName urlName = new URLName("pop3", "127.0.0.1", 3110, null, greenMailUser.getLogin(),
+                greenMailUser.getPassword());
+        try {
+            Store store = session.getStore(urlName);
+            store.connect();
+            return store;
+        } catch (MessagingException e) {
+            log.error("Error when connecting to the Email store! ", e);
+            throw new ESBMailTransportIntegrationTestException("Error when connecting to the Email store! ", e);
+        }
+    }
+
+    /**
+     * Connecting to the imap mail store
+     * @return imap store connection
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    private static Store getImapConnection() throws ESBMailTransportIntegrationTestException {
+        Properties props = new Properties();
+        Session session = Session.getInstance(props);
+        URLName urlName = new URLName("imap", "127.0.0.1", 3143, null, greenMailUser.getLogin(),
+                greenMailUser.getPassword());
+        try {
+            Store store = session.getStore(urlName);
+            store.connect();
+            return store;
+        } catch (MessagingException e) {
+            log.error("Error when connecting to the Email store! ", e);
+            throw new ESBMailTransportIntegrationTestException("Error when connecting to the Email store! ", e);
+        }
+    }
+
+    /**
+     * Check mail folder for an email using subject
+     * @param emailSubject
+     * @param folder
+     * @param protocol
+     * @return
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    public static boolean isMailReceivedBySubject(String emailSubject, String folder, String protocol)
+            throws ESBMailTransportIntegrationTestException {
+        boolean emailReceived = false;
+        Folder mailFolder;
+        Store store;
+        if (protocol.equals("pop3")) {
```

Comment:
Can be protocol null? Shall we swap the two, for the safe side?

Reply:
done